qui/tray/domains.py

```diff
@@ -144,6 +145,39 @@ def perform_shutdown(self, *_args, **_kwargs):
                            self.vm.name, str(ex)))
 
 
+class RestartItem(Gtk.ImageMenuItem):
+    ''' Restart menu Item. When activated shutdowns the domain and
+    then starts it again. '''
+
+    def __init__(self, vm, app, icon_cache):
+        super().__init__()
+        self.vm = vm
+        self.app = app
+
+        img = Gtk.Image.new_from_pixbuf(icon_cache.get_icon('restart'))
+
+        self.set_image(img)
+        self.set_label(_('Restart'))
+        self.restart_thread = None
+
+        self.connect('activate', self.restart)
+
+    def restart(self, *_args, **_kwargs):
+        asyncio.ensure_future(self.perform_restart())
+
+    async def perform_restart(self):
+        try:
+            self.vm.shutdown()
+            while self.vm.is_running():
+                await asyncio.sleep(1)
+            subprocess.Popen(['qvm-start', self.vm.name])
+        except exc.QubesException as ex:
```

Comment:
Since you call `qvm-start` as a subprocess, it will be `CalledProcessError` on start, plus you need to capture stderr to get an actual message.